app/src/main/java/xyz/malkki/neostumbler/ui/screens/MapScreen.kt

```diff
@@ -193,6 +193,16 @@ fun MapScreen(mapViewModel: MapViewModel = viewModel()) {
 
                             fillManager.value = FillManager(mapView, map, style, LocationComponentConstants.SHADOW_LAYER, null)
                         }
+
+                        val target = if (mapViewModel.mapCenter.value.isOrigin()) {
+                            latestReportPosition.value
+                        } else {
+                            mapViewModel.mapCenter.value
+                        }
+                        map.cameraPosition = CameraPosition.Builder()
+                            .target(target?.asMapLibreLatLng())
+                            .zoom(mapViewModel.zoom.value)
+                            .build()
```

Comment:
This has to be done in the `update()` function, because the latest report position is loaded asynchronously and it might not be available when the view is initialized

Reply:
In my testing, this was never a problem, the latestReportPosition was always available. But sure, I’ll change it back to update().